lib/mutant/mutator/node/send.rb

```diff
@@ -117,6 +118,22 @@ def emit_receiver_selector_mutations
             .each(&method(:emit_selector))
         end
 
+        # Emit mutation for `#dig`
+        #
+        # - Mutates `foo.dig(a, b)` to `foo.fetch(a).dig(b)`
+        # - Mutates `foo.dig(a)` to `foo.fetch(a)`
+        #
+        # @return [undefined]
+        def emit_dig_mutation
+          return unless selector.equal?(:dig) && arguments.any?
+
+          fetch_mutation = s(:send, receiver, :fetch, arguments.first)
```

Comment:
Hm ok I think I can agree with that preference. As devils advocate though: I think multiple left hand assignment might pack a bit more potential for surprise since it depends on the type being an array in this case. Even if arguments became an enumerable instance which behaved like an array this syntax would change:

```ruby
require 'concord'

class Arguments
  include Enumerable, Concord.new(:nodes)

  def each(&blk)
    nodes.each(&blk)
  end
end

arguments = [1,2,3]

# interface
arguments.first   # => 1
arguments.drop(1) # => [2, 3]

# mlhs
a1, *b1 = arguments
a1 # => 1
b1 # => [2, 3]

arguments = Arguments.new([1,2,3])

# interface
arguments.first   # => 1
arguments.drop(1) # => [2, 3]

# mlhs
a2, *b2 = arguments
a2 # => #<Arguments nodes=[1, 2, 3]>
b2 # => []
```